samples/wifi/provisioning/common/wifi_config.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <string.h>
+#include <zephyr/zephyr.h>
+#include <zephyr/settings/settings.h>
+
+#include "wifi_provisioning.h"
+#include "wifi_prov_internal.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(wifi_prov, CONFIG_WIFI_PROVISIONING_LOG_LEVEL);
+
+#define WIFI_CONFIG_KEY   "wifi_prov/wifi_config"
+
+K_MUTEX_DEFINE(prov_lock);
+static bool prov_state;
+static struct wifi_config config;
+static bool save_in_ram_only;
+
+struct direct_immediate_value {
+	size_t len;
+	void *dest;
+	uint8_t fetched;
+};
+
+static int direct_loader_immediate_value(const char *name, size_t len,
+					 settings_read_cb read_cb, void *cb_arg,
+					 void *param)
+{
+	const char *next;
+	size_t name_len;
+	int rc;
+	struct direct_immediate_value *one_value =
+					(struct direct_immediate_value *)param;
+
+	name_len = settings_name_next(name, &next);
+
+	if (name_len == 0) {
+		if (len == one_value->len) {
+			rc = read_cb(cb_arg, one_value->dest, len);
```

Comment:
legibility: looks like `&config` might be used as destination - this handler is used only to populate this one value.

Reply:
The wifi subsystem is under development and may have a lot of changes. The idea is that we may have another data type (maybe used NCS globally) to store the wifi credentials in the future, and keeping the settings-related part data type-independent makes possible adoption easier.